_posts/2021-01-10-boa-release-11.md

```diff
@@ -0,0 +1,66 @@
+        <section class="col-md-4" style="display: none" id="old-versions"></section>
+</div>
+
+## Regress
+
+In this release Boa switched from its own implementation (wrapping `regex`) to the [`regress`](https://github.com/ridiculousfish/regress) engine for regular expressions. Regress is a crate aiming to implement ECMAScript compliant regular expressions and Boa makes use (and contributes back to) that.
+
+While Regress is not 100% spec compliant this is something which is being worked on, also the switch gave us quite a performance boost in our [benchmarks](https://boa-dev.github.io/boa/dev/bench/) we're seeing almost 6X faster execution.
+
+<picture>
+    <source srcset="/images/2021-09-11/regex-bench-dark.png" media="(prefers-color-scheme: dark)">
+    <img src="/images/2021-09-11/regex-bench-white.png">
+</picture>
+
+The above image shows a big drop in the middle of the graph, above `fb1b8d5` is where we switched over. Conformance went from 19.01% to 18.99% and introduced some panics, however many of those have since been fixed.
+
+## Iterating over bytes
+
+Previously the lexer iterated over Unicode chars (u32 code points), this wasn't strictly neccesary for Boa and we have instead changed the lexer to work over bytes (u8). Iterating over bytes rather than chars is inherently much faster, non-ascii UTF8 bytes are all >=128, and we might only really care about those being correct when parsing idents. This is standard practise amongs lexical analyzers and even browsers read source code byte-by-byte, using the rules of the language's syntax to convert the source text into tokens.
+
+This was worked on by @jevancc and his [changes](https://github.com/boa-dev/boa/pull/915/files) have improved performance overall.
+
+## Embedding examples
+
+We are still working on what the public API should look like, some of these decisions are driven by feedback and the constant changing way which Boa works.  
+Not only you can run Boa against javascript today you can also embed your own objects and classes into the engine before it's ran.
+
+We've offered an [example](https://github.com/boa-dev/boa/blob/master/boa/examples/classes.rs) to show how a class can be constructed in Rust then added to the environment before executing your scripts.
+This should offer a great insight into how you can interop Rust with JavaScript by using Boa.
+
+If there are any examples you would like added, please raise an issue on the main repository.
```

Comment:
We are missing some examples until https://github.com/boa-dev/boa/pull/634 (or an equivalent PR) get merged. Should we add some code examples here? Or at least mention that using a `Context` we can register global functions, global classes and so on.

We could mention the `Context::register_global_function()`, `Context::register_global_class()` and `Context::register_global_property()` functions, to make our users understand how to extend Boa.

Reply:
Yeah I like the sound of that